vms/platformvm/blocks/builder/builder.go

```diff
@@ -245,9 +245,15 @@ func (b *builder) ResetBlockTimer() {
 // (i.e. within local time plus [MaxFutureStartFrom]).
 func (b *builder) dropExpiredStakerTxs(timestamp time.Time) {
 	minStartTime := timestamp.Add(txexecutor.SyncBound)
-	for b.Mempool.HasStakerTx() {
-		tx := b.Mempool.PeekStakerTx()
-		startTime := tx.Unsigned.(txs.Staker).StartTime()
+	iter := b.Mempool.GetTxIterator()
+	for iter.Next() {
+		tx := iter.Value()
+		stakerTx, ok := tx.Unsigned.(txs.Staker)
+		if !ok {
+			continue
+		}
+
+		startTime := stakerTx.StartTime()
```

Comment:
This is probably slower than the pre-sorted version with mempool double tx container (decision and stakers). It may be not be a big deal.
What I am more concerned with is what happen if a stakerTx gets delayed in mempool so much that its startTime was in the future at the insertion in mempool and becomes older than currentChainTime.
The tx would not pass verification anymore and should be dropped from the mempool.
I wonder if we duly evict such a tx from the mempool.
Note that this may be a problem already today, but it may get exhacerbated by the proposed change, in a heavy load situation.